### PR TITLE
Add weight decay to SGD

### PR DESCRIPTION
**Pull request by wozeparrot**
This is needed for #873 and maybe one other mlperf model

This matches the pytorch implementation.

There is a strange bug? I'm trying to fix with two specific high lr tests, `test_multistep_sgd_high_lr_wd` and `test_multistep_sgd_high_lr_nesterov_momentum_wd` which seem to have corrupted/zeroed weights on all backends. If I drop the lr for these two tests to 9 it does work but that doesn't really fix the problem.

### CODE REVIEW

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/nn/optim.py                       3               3
------------------------------------------------------------
lines added in the tinygrad folder: 0
```

**Comment by wozeparrot** [comment]
I think I may have just picked the worst possible hyperparams that perfectly cancel out weights, doesn't explain why it won't match torch tho.

**Comment by geohot** [comment]
Meh, LGTM
